pkg/rules/slp084.go

```
@@ -0,0 +1,111 @@
+func (SLP084) DefaultSeverity() Severity { return SeverityWarn }
+func (SLP084) Description() string {
+	return "useEffect may need cleanup - add return statement to clean up event listeners, timers, etc."
+}
+
+var (
+	// Matches useEffect that adds event listeners without cleanup
+	slp084EventListenerPattern = regexp.MustCompile(`(?i)(addEventListener|addEventListener\s*\([^)]*['"](?:click|scroll|resize|beforeunload|hashchange|popstate)['"])`)
+	// Matches useEffect that sets timer without cleanup
+	slp084TimerPattern = regexp.MustCompile(`(?i)(setTimeout|setInterval)\s*\(`)
+	// Matches useEffect that sets state after async without cleanup guard
+	slp084AsyncStatePattern = regexp.MustCompile(`(?i)await\s+\w+.*[\s\S]*?\.then\s*\(\s*\w+\s*=>\s*\w+\s*=\s*`)
```

golangci-lint (2.11.4)

[error] var slp084AsyncStatePattern is unused

(unused)